tensorflow/python/ops/image_ops_impl.py

```diff
@@ -1179,6 +1180,46 @@ class ResizeMethod(object):
   MITCHELLCUBIC = 'mitchellcubic'
 
 
+def _check_equal_input_output_dtypes(func):
+  """This decorator issue will issue a warning if the input dtype and output
+  dtype are different. Help to prevent silent dtype conversion from integer to
+  floating point format."""
+
+  def wrapper(*args, **kwargs):
+    try:
+      check_dtype = _check_equal_input_output_dtypes._fns_flagged[func.__name__]
+    except KeyError:
+      _check_equal_input_output_dtypes._fns_flagged[func.__name__] = True
+      check_dtype = True
+
+    if check_dtype:
+        if args:
+            input_dtype = args[0].dtype
+        else:
+            input_dtype = kwargs["images"].dtype
+
+    output = func(*args, **kwargs)
+
+    if check_dtype and (input_dtype != output.dtype):
+        _check_equal_input_output_dtypes._fns_flagged[func.__name__] = False
+        logging.warning(
+          "The operation `{func_name}` has silently converted the "
+          "data type from `{input_dtype}` to `{output_dtype}`. "
+          "This might have an adverse effect on data numerical "
+          "range.".format(
+              func_name=func.__name__,
+              input_dtype=input_dtype,
+              output_dtype=output.dtype
```

Comment:
Can we make this print the line number of the call to the function which triggered the warning? That would mean looking back in the stacktrace a few frames up ("a few" meaning how every many needed. If this utility function is used by different functions (eventually) the number of frames to unroll for this might have to be a parameter).